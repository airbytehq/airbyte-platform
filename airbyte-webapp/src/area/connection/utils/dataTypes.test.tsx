import { getDataType } from "./dataTypes";

describe("getDataType", () => {
  it.each([
    "string",
    "date",
    "timestamp_with_timezone",
    "timestamp_without_timezone",
    "datetime",
    "integer",
    "big_integer",
    "number",
    "big_number",
    "array",
    "object",
    "untyped",
    "union",
    "boolean",
  ])("should translate %s", (type) => {
    const current = getDataType({ type });
    expect(current).toMatchSnapshot();
    expect(current).not.toBe(type);
  });

  describe("union type", () => {
    it("should use union type if oneOf is set", () => {
      expect(getDataType({ type: "string", oneOf: [] })).toBe("airbyte.datatype.union");
    });
    it("should use union type if anyOf is set", () => {
      expect(getDataType({ type: "string", anyOf: [] })).toBe("airbyte.datatype.union");
    });
    it("should use union type if anyOf and oneOf is set", () => {
      expect(getDataType({ type: "string", anyOf: [], oneOf: [] })).toBe("airbyte.datatype.union");
    });
  });

  it("should return untyped when given no values", () => {
    // @ts-expect-error Using this method internally should demand a type
    expect(getDataType({})).toBe("airbyte.datatype.unknown");
  });

  describe("translation priority", () => {
    it("should use airbyte_type over format or type", () => {
      expect(getDataType({ type: "string", format: "date", airbyte_type: "big_number" })).toBe(
        "airbyte.datatype.big_number"
      );
    });
    it("should use format over type", () => {
      expect(getDataType({ type: "string", format: "date" })).toBe("airbyte.datatype.date");
    });
    it("should use oneOf/anyOf over everything else", () => {
      expect(getDataType({ type: "string", format: "date", airbyte_type: "big_number", oneOf: [] })).toBe(
        "airbyte.datatype.union"
      );
    });
  });
});
