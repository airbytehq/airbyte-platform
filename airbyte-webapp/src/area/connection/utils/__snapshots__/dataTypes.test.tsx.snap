// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDataType should translate array 1`] = `"airbyte.datatype.array"`;

exports[`getDataType should translate big_integer 1`] = `"airbyte.datatype.big_integer"`;

exports[`getDataType should translate big_number 1`] = `"airbyte.datatype.big_number"`;

exports[`getDataType should translate boolean 1`] = `"airbyte.datatype.boolean"`;

exports[`getDataType should translate date 1`] = `"airbyte.datatype.date"`;

exports[`getDataType should translate datetime 1`] = `"airbyte.datatype.datetime"`;

exports[`getDataType should translate integer 1`] = `"airbyte.datatype.integer"`;

exports[`getDataType should translate number 1`] = `"airbyte.datatype.number"`;

exports[`getDataType should translate object 1`] = `"airbyte.datatype.object"`;

exports[`getDataType should translate string 1`] = `"airbyte.datatype.string"`;

exports[`getDataType should translate timestamp_with_timezone 1`] = `"airbyte.datatype.timestamp_with_timezone"`;

exports[`getDataType should translate timestamp_without_timezone 1`] = `"airbyte.datatype.timestamp_without_timezone"`;

exports[`getDataType should translate union 1`] = `"airbyte.datatype.union"`;

exports[`getDataType should translate untyped 1`] = `"airbyte.datatype.untyped"`;
